cw_core/lib/node.dart

```diff
@@ -207,19 +214,26 @@ class Node extends HiveObject with Keyable {
   }
 
   Future<bool> requestNodeWithProxy() async {
-    if (!isValidProxyAddress /* && !Tor.instance.enabled*/) {
+    if (!isValidProxyAddress) {
       return false;
     }
 
     String? proxy = socksProxyAddress;
 
-    if ((proxy?.isEmpty ?? true) && Tor.instance.enabled) {
+    // we have to be careful here because if Tor.instance.port is called and Tor isn't enabled
+    // we'll create a new tor instance that just eats up memory
+    // we initialize tor in the tor view model first so that this doesn't happen
+    if (!Tor.instance.enabled || Tor.instance.port == -1) {
+      return false;
+    }
```

Comment:
this neglects the case where the user has an external tor running and added his own proxy address